### PR TITLE
Chris/rpc v0.4 simulate transaction

### PR DESCRIPTION
**Pull request by CHr15F0x**
Fixes: #1276 

Essentially the same as `v0.3::simulate_transaction` except for `s/code_address/class_hash/` in function execution type + `s/SKIP_EXECUTE/SKIP_FEE_CHARGE/`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,14 +186,19 @@ This can be used to interact with a custom Starknet gateway, or to use a gateway
...
 
 - the `v0.3.0` API is exposed on the `/` and `/rpc/v0.3` path
+- the `v0.4.0` API is exposed on the `/` and `/rpc/v0.4` path
 - the pathfinder extension API is exposed on `/rpc/pathfinder/v0.1`
 
 Note that the pathfinder extension is versioned separately from the Starknet specification itself.
 
+### API `v0.4.0`
+
+`starknet_simulateTransactions` currently treats the `SKIP_FEE_CHARGE` flag as permamently enabled.
```
Comment (Mirko-von-Leipzig) [inline README.md:200]: I don't think this is a good idea. If we cannot currently support the flag setting then we should throw an error instead?

File: crates/rpc/src/v03/method/simulate_transaction.rs [hunk 2]
```diff
@@ -406,18 +406,10 @@ mod tests {
                 {
                     "contract_address_salt": "0x46c0d4abf0192a788aca261e58d7031576f7d8ea5229f452b0f23e691dd5971",
                     "max_fee": "0x0",
-                    "signature": [
```
[the remark this answers is not in the document]
Comment (Mirko-von-Leipzig) [reply crates/rpc/src/v03/method/simulate_transaction.rs:409]: Ah yeah I was wondering about that too but forgot

[the remark this answers is not in the document]
Comment (CHr15F0x) [reply crates/rpc/src/v03/method/simulate_transaction.rs:409]: Guys these fields are duplicated and overwritten in the macro by their duplicates that follow :detective:

File: README.md (hunk 1 above)
Comment (Mirko-von-Leipzig) [reply README.md]: Fixed

**Review by Mirko-von-Leipzig — APPROVED** [review]

File: crates/rpc/src/v03/method/simulate_transaction.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (sergey-melnychuk) [reply crates/rpc/src/v03/method/simulate_transaction.rs:409]: That's weird. Thanks for cleaning up the duplicates!

**Pull request merged**